System/PathFinder.cpp

```diff
@@ -182,70 +171,57 @@ namespace RTE {
 		RTEAssert(g_SceneMan.GetScene(), "Scene doesn't exist or isn't loaded when recalculating PathFinder!");
 
 		// Update all the costs going out from each node
-		for (const std::vector<PathNode *> &nodeEntry : m_NodeGrid) {
-			for (PathNode *pathNode : nodeEntry) {
-				UpdateNodeCosts(pathNode);
-				// Should reset the updated flag since we're about to reset the pather
-				pathNode->IsUpdated = false;
+		std::for_each(
+			std::execution::par_unseq,
+			m_NodeGrid.begin(),
+			m_NodeGrid.end(),
+			[this](PathNode &node) {
+				UpdateNodeCosts(&node);
 			}
-		}
+		);
+
+		// Fix up the left-and-up connections
+		std::for_each(
+			std::execution::par_unseq,
+			m_NodeGrid.begin(),
+			m_NodeGrid.end(),
+			[](PathNode &node) {
+				if (node.Right) { node.Right->LeftMaterial = node.RightMaterial; }
```

Comment:
I think this is kinda confusingly written as-is, and would be a lot clearer if it just updated the PathNode's materials directly instead of updating neighbours i.e.
```cpp
std::for_each(
	std::execution::par_unseq,
	nodeVec.begin(),
	nodeVec.end(),
	[this](int nodeId) {
		PathNode &node = m_NodeGrid[nodeId];
		if (node.Left) { node.LeftMaterial = node.Left->RightMaterial; }
		if (node.Up) { node.UpMaterial = node.Up->DownMaterial; }
		if (node.DownLeft) { node.DownLeftMaterial= node.DownLeft->UpRightMaterial; }
		if (node.LeftUp) { node.LeftUpMaterial = node.LeftUp->RightDownMaterial; }
	}
);
```
Or, if you want ternaries
```cpp
std::for_each(
	std::execution::par_unseq,
	nodeVec.begin(),
	nodeVec.end(),
	[this](int nodeId) {
		PathNode &node = m_NodeGrid[nodeId];
		node.LeftMaterial = node.Left ? node.Left->RightMaterial : node.LeftMaterial;
		node.UpMaterial = node.Up ? node.Up->DownMaterial : node.UpMaterial;
		node.DownLeftMaterial = node.DownLeft ? node.DownLeft->UpRightMaterial : node.UpMaterial;
		node.LeftUpMaterial = node.LeftUp ? node.LeftUp->RightDownMaterial : node.LeftUpMaterial;
	}
);
```
Unless there's something I'm missing that would make this not work?

Reply:
That's broken. We need to update the nodes that are down-and-to-our-left, otherwise the connections at the boundaries are broken.